va/va.h

```diff
@@ -4174,6 +4174,24 @@ VAStatus vaEndPicture(
     VAContextID context
 );
 
+/**
+ * Same behavior with vaEndPicture except a sync fd list for synchronizations.
+ * if sync_num = 0 or sync_fds == NULL, the behavior should be same with vaEndPicture.
+ * if sync_num >= 1, the sync_fds[0] should be the fence out, this fd(or the fence
+ * behind this fd)  will be overridden by this call, and will be signaled to indicate
+ * current frame finishing.
+ * sync_fds[1] to sync_fds[1 ~ sync_num-1] is fence in, current HW execution
+ * will be blocked until all these fences are signaled.
+ * these fence fds is file descriptor of dma_fence.
+ */
+
+VAStatus vaEndPicture2(
+    VADisplay dpy,
+    VAContextID context,
+    int32_t *sync_fds,
```

Comment:
Is there any advantage in bundling fences out and in vs e.g.

```
   int32_t *sync_fd_out,     // Out. Can be nullptr.
   int32_t* sync_fds_in,     // In. Can be nullptr.
   int32_t sync_fds_in_num,  // In. Can be 0.
```

Or something like that?

Reply:
both ok to me, but now , it is just a PoC, and the implementation is ready here:
https://github.com/intel/media-driver/pull/1803

in the future:
1. libva should support both linux and windows, so, we will have some other interfaces to hide the dma fence, unify the syn definition in future.
2. decouple the fence in and out to too parameters as your mentioned.